### DISCUSSION TITLE
Please ask questions!

### DISCUSSION THREAD

**Discussion by BurntSushi** [rust-lang/regex#1073] [Announcements]
There are no stupid questions.

I encourage you to ask questions and I'll do my best to help answer them or direct you to a place where you might get better help. I'm happy to answer questions about the code, its design, performance, use of its API, philosophy or pretty much anything. In general I would prefer if "general" regex questions (e.g., "how do I write a regex to match an email address") get asked elsewhere, but if your question has some element that is specific to this library then it is most welcome here.

To ask a question, [please open a new one in the Q&A category](https://github.com/rust-lang/regex/discussions/new?category=q-a) so that it can be tracked individually. It will also make it more accessible to others for future reference. :-)

**Comment by johnmcilwain** [comment]
Hi, (rust beginner here) - hope this isn't a dumb question :-)
I have always wondered why there is a separate areas for regex and bytes (strings vs bytes?)?
Since rust supports generics, why 2 whole ?modules?
thanks!

**Reply by BurntSushi** [reply]
Whoops, sorry about that, I intended for folks to open a new question for each one. I've opened this as question for you: https://github.com/rust-lang/regex/discussions/1075

**Comment by sam0x17** [comment]
What is the current status of compile-time regex support (last I checked this was developed separately in the `regex-automata` crate) and what barriers exist currently for bringing this functionality into the main release?

Do the recent improvements in terms of what `const fn`s can do in stable Rust make it feasible to do compile-time regex via `const` machinery, and has any work been done in that direction? If not, what features would need to be added to const eval contexts to enable this?

With the recent absorption of `once_cell` into `std`, do you think there is a path forward for `regex` eventually becoming part of the rust standard library? If not, why, especially considering the fact that most languages include some sort of regex facility built into the language?

Anyway, love your work!

**Reply by BurntSushi** [reply]
Whoops, sorry about that, I intended for folks to open a new question for each one. I've opened this as question for you: https://github.com/rust-lang/regex/discussions/1076
